internal/host/hostcommands/api_vip_connectivity_check_cmd.go

```diff
@@ -32,16 +34,40 @@ func (c *apivipConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.
 		c.log.WithError(err).Errorf("failed to fetch cluster %s", host.ClusterID)
 		return nil, err
 	}
+	var commonHost common.Host
+	if err := c.db.First(&commonHost, "id = ?", host.ID).Error; err != nil {
+		c.log.WithError(err).Errorf("failed to fetch common Host %s", host.ID)
+		return nil, err
+	}
 
 	addressPart := swag.StringValue(cluster.APIVipDNSName)
 	if addressPart == "" {
 		addressPart = cluster.APIVip
 	}
-
-	apiURL := fmt.Sprintf("http://%s:22624/config/worker", addressPart)
+	var caCert *string = nil
+	poolName := "worker"
+	if host.MachineConfigPoolName != "" {
+		poolName = host.MachineConfigPoolName
+	}
+	ignitionEndpointUrl := fmt.Sprintf("http://%s:22624/config/%s", addressPart, poolName)
+	if cluster.IgnitionEndpoint != nil && cluster.IgnitionEndpoint.URL != nil {
+		caCert = cluster.IgnitionEndpoint.CaCertificate
+		url, err := url.Parse(*cluster.IgnitionEndpoint.URL)
+		if err != nil {
+			return nil, err
+		}
+		url.Path = path.Join(url.Path, poolName)
+		ignitionEndpointUrl = url.String()
+	}
```

Comment:
Consider extracting this logic to a separate function `getIgnitionURL`
* the caCert part doesn't have to be in it.

Reply:
Fixed